SIMSBIOHUB-501: Import Observations Against Sample Period

NickPhura opened the pull request:
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-501

## Description of Changes

Add import button to sample period item in sample site list (on the observations page).

Update some of the loading/disabled states, so that when you import observations from either the sample period or the observations table, it disables the controls on the page.

## Testing Notes

Importing observations from both the sample period and observations table works.
Existing controls work as they did before.

NickPhura committed:
Initial working observation import against sampling period

NickPhura committed:
Add better loading/disabled handlers

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

NickPhura committed:
Add Knip and SWC

NickPhura committed:
Update import observations

NickPhura committed:
Remove console logs

NickPhura committed:
Remove swc from API to fix unit tests

NickPhura committed:
Remove knip

NickPhura committed:
Remove gulp

NickPhura committed:
Update lock

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-501

NickPhura committed:
Merge branch 'SIMSBIOHUB-501' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-501

NickPhura committed:
Add unit tests. fix spelling

NickPhura committed:
Add tests

mauberti-bc reviewed (dismissed):
Looks good to me, and works as expected locally when uploading one of the observation files that we've been testing with.

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

NickPhura committed:
Fix merge conflicts

NickPhura committed:
run lint/format

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/import-observations/ImportObservationsButton.tsx:
```diff
@@ -0,0 +1,146 @@
+   * @param {File} file
+   * @return {*}
+   */
+  const handleImportObservations = async (file: File) => {
```
Nit: A try catch here might make this slightly more readable.

NickPhura replied:
Updated

NickPhura replied:
Had 2 nearly identical components doing the same thing, so I've merged them into one file in a new folder.

MacDeluca commented on database/package.json:
```diff
@@ -23,6 +23,7 @@
     "fix": "npm-run-all -l -s lint-fix format-fix"
   },
   "dependencies": {
+    "@swc/core": "^1.3.76",
```
:heart_eyes_cat:

NickPhura replied:
I did have to take it out of the API unfortunately, for now at least, as it was causing our unit tests to break. But working fine in the app (and database, tho it probably doesn't matter)

MacDeluca commented on api/src/services/observation-service.ts:
```diff
@@ -455,9 +471,9 @@ export class ObservationService extends DBService {
           survey_id: surveyId,
           itis_tsn: row['ITIS_TSN'] ?? row['TSN'] ?? row['TAXON'] ?? row['SPECIES'],
           itis_scientific_name: null,
```
Do we need to support additional mappings here for `itis_scientific_name`?
ie: latin_name / scientific_name / species_name

NickPhura replied:
Not sure, I'd have to confirm with Macgregor. 

Unless you know of other aliases already in use?

mauberti-bc replied:
I don't think we want to identify species based on scientific name. Although it can be used to identify a taxon, the code is preferred.

MacDeluca commented on app/src/features/surveys/observations/observations-table/import-observations/ImportObservationsButton.tsx:
```diff
@@ -11,11 +11,45 @@ import { useBiohubApi } from 'hooks/useBioHubApi';
+   */
 }
 
 export const ImportObservationsButton = (props: IImportObservationsButtonProps) => {
```
jsdoc :page_with_curl:

NickPhura replied:
added

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/SamplingSiteListMethod.tsx:
```diff
@@ -0,0 +1,51 @@
+  sampleMethod: IGetSampleMethodRecord;
+}
+
+export const SamplingSiteListMethod = (props: ISamplingSiteListMethodProps) => {
```
jsdoc 📃

NickPhura replied:
Added

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/SamplingSiteListPeriod.tsx:
```diff
@@ -0,0 +1,59 @@
+  samplePeriod: IGetSamplePeriodRecord;
+}
+
+export const SamplingSiteListPeriod = (props: ISamplingSiteListPeriodProps) => {
```
jsdoc 📃

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/SamplingSiteListSite.tsx:
```diff
@@ -0,0 +1,118 @@
+  handleCheckboxChange: (sampleSiteId: number) => void;
+}
+
+export const SamplingSiteListSite = (props: ISamplingSiteListSiteProps) => {
```
jsdoc 📃

NickPhura replied:
added

MacDeluca commented on app/src/features/surveys/observations/observations-table/import-observations/ImportObservationsButton.tsx:
```diff
@@ -37,14 +71,14 @@ export const ImportObservationsButton = (props: IImportObservationsButtonProps)
    * @return {*}
    */
   const handleImportObservations = async (file: File) => {
```
nit: try catch?

NickPhura replied:
Updated to use a regular try/catch + async/await

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/SamplingSiteListSite.tsx:
```diff
@@ -0,0 +1,118 @@
+              typography: 'body2'
+            }
+          }}>
+          {sampleSite.sample_methods?.map((sampleMethod) => {
```
console warning here for missing key

NickPhura replied:
Added

MacDeluca commented on app/src/features/surveys/observations/sampling-sites/list/SamplingSiteListMethod.tsx:
```diff
@@ -0,0 +1,51 @@
+      />
+      <List disablePadding>
+        {sampleMethod.sample_periods?.map((samplePeriod) => {
+          return <SamplingSiteListPeriod samplePeriod={samplePeriod} />;
```
missing key

NickPhura committed:
Code review comments:
Add JSDoc.
Remove duplicate import observations component.
Add missing sample list keys
Update prettier deprecated option.
Minor css tweak to fix re-center button on sample site map.

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

NickPhura committed:
Fix refresh after importing observations via sample period.
Disable observation import button if no file selected

NickPhura committed:
Fix bad tests.
Fix css kebab case warning

MacDeluca reviewed (approved):
:cactus:
